src/operations/booleans/trees/Tree.js

```diff
@@ -31,13 +31,17 @@ Tree.prototype = {
   },
 
   addPolygons: function (polygons) {
-    let _this = this
-    let polygontreenodes = polygons.map(function (p) {
-      return _this.polygonTree.addChild(p)
-    })
+    let polygontreenodes = new Array(polygons.length)
+    for (let i = 0; i < polygons.length; i++) {
+      polygontreenodes[i] = this.polygonTree.addChild(polygons[i])
```

Comment:
any reason for the need to pre-init the polygontreenodes ? 
(speed benefit of for vs map perhaps ?)

Reply:
correct.